kai/rpc_server/server.py

```diff
@@ -506,37 +508,58 @@ def simple_chat_message(msg: str) -> None:
         seed_tasks: list[Task] = []
 
         params.incidents.sort()
-        for incident in params.incidents:
-
-            class_to_use = AnalyzerRuleViolation
-            if "pom.xml" in incident.uri:
-                class_to_use = AnalyzerDependencyRuleViolation
-
-            # handle windows paths
-            uri_path = urlparse(incident.uri).path
-            if platform.system() == "Windows":
-                uri_path = uri_path.removeprefix("/")
-
-            seed_task = class_to_use(
-                file=str(Path(uri_path).absolute()),
-                line=incident.line_number,
-                column=-1,  # Not contained within report?
-                message=incident.message,
-                priority=0,
-                incident=Incident(**incident.model_dump()),
-                violation=Violation(
-                    id=incident.violation_name or "",
-                    description=incident.violation_description or "",
-                    category=incident.violation_category,
-                    labels=incident.violation_labels,
-                ),
-                ruleset=RuleSet(
-                    name=incident.ruleset_name,
-                    description=incident.ruleset_description or "",
-                ),
-            )
-
-            seed_tasks.append(seed_task)
+        grouped_incidents_by_files = [
+            list((g)) for _, g in groupby(params.incidents, key=attrgetter("uri"))
+        ]
+        for incidents in grouped_incidents_by_files:
+
+            # group incidents by violation
+            grouped_violations = [
+                list((g))
```

Comment:
Any reason for the double parenthesis in `list((g))`?

Reply:
Turns out you don't they were utterly superflous. 

I thought you needed them, but you don't removing in the same PR